src/commands/releases.rs

```diff
@@ -507,6 +517,63 @@ fn execute_set_commits<'a>(
     Ok(())
 }
 
+fn execute_set_manual_commits<'a>(ctx: &ReleaseContext<'_>, matches: &ArgMatches<'a>) -> Result<(), Error> {
+    let version = matches.value_of("version").unwrap();
+    let org = ctx.get_org()?;
+
+    if let Ok(projects) = ctx.get_projects(matches) {
+        ctx.api.new_release(
+            &org,
+            &NewRelease {
+                version: version.into(),
+                projects,
+                ..Default::default()
+            },
+        )?;
+    }
+
+    // Get the commit of the most recent release.
+    let prev_commit = match ctx.api.get_previous_release_with_commits(org, version)? {
+        OptionalReleaseInfo::Some(prev) => prev.last_commit.map(|c| c.id).unwrap_or_default(),
+        OptionalReleaseInfo::None{} => String::new(),
+    };
+
+    // Find and connect to local git.
+    let repo = git2::Repository::open_from_env()?;
+
+    // Parse the git url.
+    let remote = repo.find_remote("origin")?;
+    let url = remote.url();   
+    let parsed = match url {
+        Some(url) => parse_git_url(url),
+        None => return Err(format_err!("Failed to parse remote url!"))
+    };
+
+    // Fetch all the commits upto the `prev_commit` or return the default (20).
+    // Will return a tuple of Vec<GitCommits> and the `prev_commit` if it exists in the git tree.
+    let (commit_log, prev_commit) = get_commits_from_git(&repo, &prev_commit)?;
+    
+    // Calculate the diff for each commit in the Vec<GitCommit>.
+    let commits = generate_patch_set(&repo, commit_log, prev_commit, &parsed)?;
+
+    for chunk in commits.chunks(50) {
```

Comment:
Magic number could be extracted and named to be more descriptive (and most likely configurable in the future).